### PR TITLE
CGAL_Qt5: include QGLViewer

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
- The code of QGLViewer 2.7.0 has been integrated into CGAL_Qt5
- The drawVisualHints() function working with recent OpenGL has been ported to it from the Polyhedron Demo for the most part
- The not working snapshot feature has been removed
- The context creation has also been integrated

All of this means that QGLViewer is no longer a dependency for the demos, because it is now intern to the Graphicsview package.

## Release Management

* Affected package(s):GraphicsView

### CODE REVIEW

**Comment by gdamiand** [comment]
Very good idea ! A nice step to simplify users life.

@maxGimeno  and @lrineau This will have conflicts with https://github.com/CGAL/cgal/pull/2835; thus maybe it is better to integrate first my PR then this one ?

**Comment by lrineau** [comment]
@
[**@gdamiand**](https://github.com/gdamiand) commented on [Mar 30, 2018, 9:53 AM GMT+2](https://github.com/CGAL/cgal/pull/2967#issuecomment-377470536 "2018-03-30T07:53:10Z - Replied by Github Reply Comments"):
> Very good idea ! A nice step to simplify users life.
> 
> @maxGimeno  and @lrineau This will have conflicts with https://github.com/CGAL/cgal/pull/2835; thus maybe it is better to integrate first my PR then this one ?

Your PR #2835 will come first (because it was ready before). But anyway I do not think there will be conflicts. There should not, because the changes of this PR #2967 only touch to QGLViewer, and nothing from CGAL code.

Or maybe in the CMake scripts. I will deal with the conflicts once we have them.

**Comment by gdamiand** [comment]
Yes I was thinking about CMake scripts. 

In my PR, I updated FindQGLViewer and some CMake lists to use a new virtual target for linking automatically with QGLViewer and the required dependencies. This should be updated according to this PR.

**Comment by maxGimeno** [comment]
> PR #2967 only touch to QGLViewer, and nothing from CGAL code.

Yes it does. It touches CMakeLists, of course, and QGLViewer's constructor changed too, for optimization purpose.

**Comment by maxGimeno** [comment]
There is none for now, it is the next feature I'll have to move from the Polyhedron demo. I thought it was based on the demo's scene so I just removed the feature, but I was wrong.

File: GraphicsView/include/CGAL/Qt/camera.h [hunk 1]
```diff
@@ -19,6 +19,7 @@
  WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
 
 *****************************************************************************/
+// SPDX-License-Identifier: GPL-3.0+
```
Comment (lrineau) [inline GraphicsView/include/CGAL/Qt/camera.h]: The license notice says:
> This file may be used under the terms of the GNU General Public License 
 versions 2.0 or 3.0 as published by the Free Software Foundation

That cannot be GPL-3.0+. We can use it under GPL-3.0 (with the `+`).

**Comment by maxGimeno** [comment]
Ok I give up. @lrineau can you help me with that license check ? I feel like I tried everything and it still won't work.

**Comment by maxGimeno** [comment]
@afabri told me that it works, no more compilation nor execution errors. I guess I should change the namespaces, replace qglviewer:: by CGAL::Qt:: and put QGLViewer in CGAL::
Do we keep the name QGLViewer ?

**Comment by afabri** [comment]
I suggest to keep the name `QGLViewer` as that makes it easier to diff with the evolving original version.

**Comment by lrineau** [comment]
We have decided to include everything in the CGAL:: namespace: the class `QGLViewer` will become `CGAL::QGLViewer`, and everything under the namespace `::qglviewer::` will be moved to `CGAL::qglviewer::`.

**Comment by lrineau** [comment]
The Travis errors seems pertinent. Ignore the CHECK line for the moment.

File: Polyhedron/demo/Polyhedron/include/Point_set_3.h [hunk 2]
```diff
@@ -561,7 +561,7 @@ namespace CGAL
 namespace Point_set_processing_3
 {
   template<typename Kernel>
-  class GetFT<::Point_set_3<Kernel> >
+  class GetFT< ::Point_set_3<Kernel> >
```
Comment (lrineau) [inline Polyhedron/demo/Polyhedron/include/Point_set_3.h]: @maxGimeno @sgiraudot Is not `Point_set_3` in the CGAL namespace?!

Comment (maxGimeno) [reply Polyhedron/demo/Polyhedron/include/Point_set_3.h]: at the beginning of the file : 
template <class Gt>
class Point_set_3 : public CGAL::Point_set_3<typename Gt::Point_3,
                                             typename Gt::Vector_3>

Comment (sgiraudot) [reply Polyhedron/demo/Polyhedron/include/Point_set_3.h]: `Point_set_3` is a class of the demo that inherits from `CGAL::Point_set_3` (for historical reasons actually, `Point_set_3` was always in the demo and `CGAL::Point_set_3` was introduced later as a generalization).

**Comment by lrineau** [comment]
Remaining Travis error:
```
[100%] Linking CXX executable periodic_3_triangulation_3_demo
CMakeFiles/periodic_3_triangulation_3_demo.dir/Viewer.cpp.o: In function `Viewer::Viewer(QWidget*)':
/home/travis/build/CGAL/cgal/Periodic_3_triangulation_3/demo/Periodic_3_triangulation_3/Viewer.cpp:(.text+0x50): undefined reference to `CGAL::QGLViewer::QGLViewer(QGLContext*, QWidget*, QFlags<Qt::WindowType>)'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [periodic_3_triangulation_3_demo] Error 1
make[1]: *** [CMakeFiles/periodic_3_triangulation_3_demo.dir/all] Error 2
```
https://travis-ci.org/CGAL/cgal/jobs/367089813#L3067

**Comment by lrineau** [comment]
There is a conflict in this PR. Once it is solved, we will test it in `integration`.

**Comment by maxGimeno** [comment]
As I cannot reproduce the errors in the testsuite, I can only hope this fixes them.

**Comment by sloriot** [comment]
Doing it right now.

**Comment by sloriot** [comment]
Done

**Comment by lrineau** [comment]
[**@lrineau**](https://github.com/lrineau) commented on [May 17, 2018, 7:23 PM GMT+2](https://github.com/CGAL/cgal/pull/2967#issuecomment-389944403 "2018-05-17T17:23:15Z - Replied by Github Reply Comments"):
> @sloriot @maxGimeno The history will have to be rewrote, to filter out the binary file `GraphicsView/src/CGAL_Qt5/qglviewer.icns` that is added by 7dc02e827a4 and then removed by bb1a8f70d6a.


[**@sloriot**](https://github.com/sloriot) commented on [May 18, 2018, 8:17 AM GMT+2](https://github.com/CGAL/cgal/pull/2967#issuecomment-390105887 "2018-05-18T06:17:40Z - Replied by Github Reply Comments"):
> Done

@sloriot I still see the file `GraphicsView/src/CGAL_Qt5/qglviewer.icns` in the history.

@maxGimeno @afabri Has one of you used a forced pushes, and cancel the new history constructed by Sébastien?

**Comment by maxGimeno** [comment]
I don't think so.

**Comment by sloriot** [comment]
81d439a  is probably responsible for that. @lrineau  Shall I fix it?

**Comment by lrineau** [comment]
Yes, please fix the history again if that is possible.

**Comment by sloriot** [comment]
On it please, don't push anything.

**Comment by sloriot** [comment]
Should be good now

**Comment by lrineau** [comment]
Successfully tested in [CGAL-4.13-Ic-21][*], and then merged.

  [*]: https://cgal.geometryfactory.com/CGAL/testsuite/results-4.13-Ic-21.shtml

**Pull request merged**